### DISCUSSION TITLE
Does the rewrite involve this? [Downloading, Codecs, UX]

### DISCUSSION THREAD

**Discussion by n-ce** [TeamNewPipe/NewPipe#11763] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

The download prompt currently only asks for video quality
can we have audio quality selection as well (not the separate one).
Like I am pretty sure most 144p video downloaders don't need 130kbps audio, so we must either smartly include a lower quality audio or let the user choose what quality they want to use.

Why this matters?
Let's take this video for a reference : https://m.youtube.com/watch?v=GWB9ApTPTv4

The 480p AV1 videostream costs 93MB
The 251 audiostream costs 142MB
so 93+142 = 235MB

More than the videostream? That's stupid from the user point of view and what about 600 audiostream? It's just 44MB.

That's a 100MB of a difference!

(Alongside the quality I also hope some transparency is there about the codecs being used)

This is the only reason I am using yt-dlp instead of NewPipe.

**Comment by ShareASmile** [comment]
This needs download refactor. After rewrite it will be looked into whether ditch old downloader in favour of a new one or not. You can follow its dedicated issue on the tracker.
